src/components/look/look-list/style-tag-list/index.tsx

```diff
@@ -0,0 +1,66 @@
+import styled from 'styled-components';
+import { useState } from 'react';
+import { message } from 'antd';
+
+import { Button } from '@src/component/atoms';
+
+import { StyleTag } from '@src/types';
+import styleTagsData from '@src/data/style-tags.json';
+import { LookService } from '@src/services';
+
+type StyleTagListProps = {
+  styleTags: StyleTag[];
+  id: number;
+};
+
+function StyleTagList({ styleTags, id }: StyleTagListProps) {
+  const [tags, setTags] = useState<StyleTag[]>([...styleTags]);
+
+  const handleCheck = (tag: StyleTag) => {
+    tags.find((t) => t.id === tag.id)
+      ? setTags(tags.filter((t) => tag.id !== t.id))
+      : setTags((prevStayleTags) => [...prevStayleTags, tag]);
+  };
```

Comment:
1. 가독성 높이기
  built in 함수들을 적절히 사용해 가독성을 높일 수 있습니다. (Array.prototype.includes, concat)
2. 중복 로직 없애기
  setTags 함수 호출이 불필요하게 중복됩니다! 중복을 최대한 제거해주시는게 좋습니다.

```suggestion
  const handleCheck = (id: number) => {
    const isSelected = selectedTagIds.includes(id);
    setSelectedTagids(selectedTagIds.concat(isSelected ? [] : id));
  };
```

Reply:
엇 그런데 이렇게 변경하면 체크 isSelected가 ture 일때 selectedTagIds에서 id를 제거하지 않아 체크 해제가 안되는데, 이건 어떻게 해결하나요??

Reply:
```suggestion
  const handleCheck = (id: number) => {
    const isSelected = selectedTagIds.includes(id);
    setSelectedTagids(
      selectedTagIds.filter(selectedTagId => selectedTagId != id).concat(isSelected ? [] : id)
    );
  };
```

앗 제가 실수했네요.
먼저 filter하고 concat해주시면 됩니다!

Reply:
d3d4728

Reply:
```js
const handleCheck = (id: number) => {
    const isSelected = selectedTagIds.includes(id);
    setSelectedTagIds(
      isSelected
        ? selectedTagIds.filter((selectedTagId) => selectedTagId !== id)
        : selectedTagIds.concat(id)
    );
  };
```
일단 이렇게 했습니다!

Reply:
filter가 반영이 안 됐네요 ㅠㅠ

Reply:
2cdaa1f